### PR TITLE
User data

### PR DESCRIPTION
**Pull request by Guptacos**
Add getter methods to the User class for various user information

### CODE REVIEW

**Comment by Guptacos** [comment]
The images branch must be merged before this one

File: spotifython/user.py [hunk 1]
```diff
@@ -19,6 +22,11 @@ class User:
     Raises:
         TypeError:  if incorrectly typed parameters are given.
         ValueError: if parameters with illegal values are given.
+
+    Unsupported:
+
+        * external_urls
+        * followers: unsupported by Spotify, see https://developer.spotify.com/documentation/web-api/reference/object-model/#followers-object
```
Comment (rmannava) [inline spotifython/user.py:29]: google style guide says to put the url on its own line of its too long, might not need the pylint disable

File: spotifython/user.py [hunk 2]
```diff
@@ -40,6 +48,71 @@ def __init__(self, session, info):
...
+
+        if status_code != 200:
+            raise utils.SpotifyError(status_code, response_json)
+
+        # Updates _raw with new values. One liner : for each key in union of
+        # keys in self._raw and response_json, takes value for key from
+        # response_json if present, else takes value for key from self._raw.
+        # TODO: this is weird notation, make a utility function for it.
+        # Especially useful since it is an action necessary for many classes.
+        self._raw = {**self._raw, **response_json}
```
Comment (rmannava) [inline spotifython/user.py]: would self._raw.update(response_json) do the same thing?

File: spotifython/user.py [hunk 3]
```diff
@@ -40,6 +48,71 @@ def __init__(self, session, info):
...
+
+        # Updates _raw with new values. One liner : for each key in union of
+        # keys in self._raw and response_json, takes value for key from
+        # response_json if present, else takes value for key from self._raw.
+        # TODO: this is weird notation, make a utility function for it.
+        # Especially useful since it is an action necessary for many classes.
+
+        # We know other is a 'User' object, so this protected access is okay
+        #pylint: disable=protected-access
+        self._raw = {**self._raw, **other._raw}
```
Comment (rmannava) [inline spotifython/user.py]: same as above

**Review by rmannava — CHANGES_REQUESTED** [review]

File: spotifython/user.py (hunk 2 above)
Comment (spenceryu) [reply spotifython/user.py]: I use this magic method across a lot of my code too. If we change this, it should be a separate PR

**Review by spenceryu — DISMISSED** [review]

File: spotifython/user.py (hunk 2 above)
Comment (Guptacos) [reply spotifython/user.py]: Yea it does see the groupchat

File: spotifython/user.py (hunk 1 above)
Comment (Guptacos) [reply spotifython/user.py:29]: It's too long even on its own line; this way at least it renders nicely with sphinx

**Pull request merged**